sources/datasource/utils.go

```diff
@@ -15,31 +15,12 @@
 package datasource
 
 import (
-	"time"
-
-	"github.com/golang/glog"
 	cadvisor "github.com/google/cadvisor/info/v1"
 )
 
-func sampleContainerStats(stats []*cadvisor.ContainerStats, resolution time.Duration) []*cadvisor.ContainerStats {
+func sampleContainerStats(stats []*cadvisor.ContainerStats) []*cadvisor.ContainerStats {
 	if len(stats) == 0 {
 		return stats
 	}
-	filteredStats := []*cadvisor.ContainerStats{}
```

Comment:
After poll_duration flag is removed we want to get exactly one sample for each time interval. There is also a possibility to add support of querying for 1 sample from time interval in cadvisor which will reduce significantly network load (see https://github.com/kubernetes/heapster/pull/512#discussion_r38791456).